src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs

```diff
@@ -103,8 +99,16 @@ public static bool IsOSPlatform(string platform!!)
             return platform.Equals("TVOS", StringComparison.OrdinalIgnoreCase);
 #elif TARGET_ANDROID
             return platform.Equals("ANDROID", StringComparison.OrdinalIgnoreCase);
-#elif TARGET_UNIX
-            return platform.Equals(s_osPlatformName, StringComparison.OrdinalIgnoreCase);
+#elif TARGET_LINUX
+            return platform.Equals("LINUX", StringComparison.OrdinalIgnoreCase);
+#elif TARGET_FREEBSD
+            return platform.Equals("FREEBSD", StringComparison.OrdinalIgnoreCase);
+#elif TARGET_NETBSD
+            return platform.Equals("NETBSD", StringComparison.OrdinalIgnoreCase);
+#elif TARGET_ILLUMOS
+            return platform.Equals("ILLUMOS", StringComparison.OrdinalIgnoreCase);
+#elif TARGET_SOLARIS
+            return platform.Equals("SOLARIS", StringComparison.OrdinalIgnoreCase);
 #else
 #error Unknown OS
```

Comment:
nit: We are deleting this build-time error message from cmake:

```cmake
message(FATAL_ERROR "Unknown platform. Cannot define PAL_UNIX_NAME, used by RuntimeInformation.")
```

which is slightly more helpful. Can we use similar message here?